site/content/docs/developer/quick-start-build.md

```diff
@@ -0,0 +1,146 @@
+---
+title: "A Quick Start Guide to Building ZAP"
+description: "This guide will get you started with running ZAP and installing add-ons from source."
+images:
+- https://www.zaproxy.org/img/docs/developer/zap-dev-build.png
+tags: 
+- guide
+- tutorial
+type: page
+date: "2020-12-04"
+addSocialPreview: true
+---
+
+Have you ever wanted to play around with the ZAP codebase but felt it was too overwhelming? This guide will walk you through building ZAP from the command line regardless of the operating system and IDE you're using. Let’s jump right into it!
+
+## Preparation
+All you need is a working local installation of [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) and [JDK 8+](https://adoptopenjdk.net/).
+
+## Clone the Repo Locally
+Create a new directory on your system and run the following command in that directory to clone the core ZAP repository.
+```bash
+git clone https://github.com/zaproxy/zaproxy.git
```

Comment:
Shouldn't people be cloning their fork as the first step? (So that origin properly points to their fork, and they don't end-up with origin/upstream the same).

Reply:
I wanted to to make it as easy as possible for people to get ZAP running from the source code.
Cloning repos complicates things and isnt necessary. But happy to have some text pointing out that if people do plan to contribute back then they should go to step X

Reply:
Well, to prevent that we have a section at the end where they point origin to their fork (using `git remote set-url origin https://github.com/USERNAME/zaproxy.git`) and upstream to the main repo. But if cloning would be easier / more clear I can update the docs. What do you suggest?

Reply:
@psiinon if these are developer docs isn't the assumption that they're contributing?

Or should we put this under a heading other than "developer"? (`/docs/desktop/run-from-source/`?)

Reply:
Tricky - I want us to have a new section for ZAP internals - this is just the start of that section. But you are right, this page isnt just for ZAP devs. So open to suggestions ...
EG would love to migrate a load of this content: https://github.com/zaproxy/zaproxy/wiki/InternalDetails

Reply:
Well we all know I over think these things so maybe we leave it as-is since there is a chunk later in the doc that addresses origin/upstream.

If we end-up with lots of questions or devs with problems we can always revisit.

Reply:
Being developer does not imply being a contributor, e.g. you might want to create a custom version of ZAP for yourself. (There are many forks of zaproxy with custom changes that are not contributed back, offline changes should be a higher number I guess.)

Reply:
Good point. Okay let's leave it as-is for now.